lib/Constants.js

```diff
@@ -366,9 +366,27 @@ const RequestStatus = {
 const SearchKeys = {
   Requests: {
     ID: (q, r) => r.id.toString() === q,
-    LOCATION: (q, r) => r.location.description.indexOf(q) !== -1,
-    REQUESTER: (q, r) => r.requestedBy.name.indexOf(q) !== 1,
-    STATUS: (q, r) => r.status.indexOf(q) !== 1,
+    LOCATION: (q, r) => {
+      if (r.location === null) {
+        return q === '';
+      }
+      const qLower = q.toLowerCase();
+      const rLower = r.location.description.toLowerCase();
+      return rLower.indexOf(qLower) !== -1;
+    },
+    REQUESTER: (q, r) => {
+      if (r.requestedBy === null) {
```

Comment:
same here: guard against falsy `r.requestedBy.name`.  in addition, we should change the checks for `r.location` and `r.requestedBy` to catch all falsy values (`null`, `undefined`, empty string).